clang_delta/RemoveUnusedEnumMember.cpp

```diff
@@ -52,14 +52,20 @@ bool RemoveUnusedEnumMemberAnalysisVisitor::VisitEnumDecl(EnumDecl *ED)
   if (ConsumerInstance->isInIncludedFile(ED) || ED != ED->getCanonicalDecl())
     return true;
 
+  /* Make it backward compatible where --to-counter is unset. */
+  if (ConsumerInstance->ToCounter == -1)
+    ConsumerInstance->ToCounter = ConsumerInstance->TransformationCounter;
+
   for (EnumDecl::enumerator_iterator I = ED->enumerator_begin(),
        E = ED->enumerator_end(); I != E; ++I) {
     if (!(*I)->isReferenced()) {
       ConsumerInstance->ValidInstanceNum++;
-      if (ConsumerInstance->ValidInstanceNum ==
-          ConsumerInstance->TransformationCounter) {
-        ConsumerInstance->TheEnumIterator = I;
-        ConsumerInstance->TheEnumDecl = ED;
+      if (ConsumerInstance->ValidInstanceNum >=
+          ConsumerInstance->TransformationCounter
+          && ConsumerInstance->ValidInstanceNum <=
+             ConsumerInstance->ToCounter) {
+        TransAssert(ED && "NULL TheEnumDecl!");
```

Comment:
Seems this assert is not necessary, because ```ED``` has actually be used. If it's a nullptr, we would fail before we hit this assert.

Reply:
You are right.